### DISCUSSION TITLE
Has anyone tested Fuse for large datasets?

### DISCUSSION THREAD

**Discussion by hhimanshu** [krisk/Fuse#577] [Q&A] [answered]
Hello,

I have almost 3 million records, and I wanted to find out if someone has tested the library for large indexes.

Any help is greatly appreciated
Thanks

**Comment by krisk** [comment, marked as answer]
3M is too much. I do not recommend storing 3M records on the client. Additionally, note that Fuse.js creates an index, and that index lives in memory. For 3M record, that's a lot.

Technically, if you're running Fuse.js on the server, it is doable, but at that point, it also depends not only on the number of records but, the _size_ of those records. Still, at this point, you may as well go with a distributed solution.
